### ISSUE TITLE
Certificate generated as part of a challenge with the acme_certificate module does not respect the content of the CSR file

### ISSUE THREAD

**Issue by GuillaumeBernard3535** [ansible-collections/community.crypto#723]
##### SUMMARY
When we challenge a certificate request with SAN, the generated certificate does not respect the CNAME or order of the SAN. Randomly, one of the aliases in the SAN is mistaken for CNAME and the order of the aliases in the SAN is mixed.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

Module acme_certificate
DIGICERT has identified a bug in file /library/module_utils/acme/backend_cryptography.py in the get_csr_identifiers function
It proposes to replace the definition of the following variable identifiers by changing its type
identifiers = set([]) ---> identifiers = []
and to use append method instead of add method in order to order the list of DNS properties

##### ANSIBLE VERSION

```paste below
ansible [core 2.14.10]
python version = 3.9.18 (main, Sep  7 2023, 00:00:00) [GCC 11.4.1 20230605 (Red Hat 11.4.1-2)] (/usr/bin/python3)
jinja version = 3.1.2
libyaml = True
```

##### COLLECTION VERSION

```paste below
Collection       Version
---------------- -------
community.crypto 2.18.0
```

##### CONFIGURATION

```paste below
CACHE_PLUGIN(env: ANSIBLE_CACHE_PLUGIN) = jsonfile
CACHE_PLUGIN_CONNECTION(env: ANSIBLE_CACHE_PLUGIN_CONNECTION) = /runner/artifacts/c1aa00ba-2797-46ea-85c8-c181e1f7285b/fact_cache
COLLECTIONS_PATHS(env: ANSIBLE_COLLECTIONS_PATHS) = ['/home/gs6015/migration/creation_certificat/.collections']
CONFIG_FILE() = /home/gs6015/migration/creation_certificat/ansible.cfg
DEFAULT_CALLBACK_PLUGIN_PATH(env: ANSIBLE_CALLBACK_PLUGINS) = ['/runner/artifacts/c1aa00ba-2797-46ea-85c8-c181e1f7285b/callback']
DEFAULT_ROLES_PATH(env: ANSIBLE_ROLES_PATH) = ['/home/gs6015/migration/creation_certificat/.roles']
DEFAULT_STDOUT_CALLBACK(env: ANSIBLE_STDOUT_CALLBACK) = awx_display
DEFAULT_VAULT_PASSWORD_FILE(env: ANSIBLE_VAULT_PASSWORD_FILE) = /home/gs6015/migration/creation_certificat/.ansible_vault
HOST_KEY_CHECKING(env: ANSIBLE_HOST_KEY_CHECKING) = False
RETRY_FILES_ENABLED(env: ANSIBLE_RETRY_FILES_ENABLED) = False
```

##### OS / ENVIRONMENT

Linux 4.18.0-408.el8.x86_64 #1 SMP Mon Jul 18 17:42:52 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

##### STEPS TO REPRODUCE

A CSR file with :
- Common name :
test-123.domain.fr
- SANs :
test-123.domain.fr
test-123-1.domain.fr
test-123-2.domain.fr

An ACME URL: mine is https://one.digicert.com/mpki/api/v1/acme/v2/directory

```yaml
### ACME Etape 1 ################################################################################
- name: Récupération du challenge DIGICERT
  community.crypto.acme_certificate:
    select_crypto_backend: cryptography
    acme_version: 2
    acme_directory: "{{ acme_dir_url }}"
    account_key: "{{ data_dir }}/{{ account_key_name }}.pem"
    account_key_passphrase: "{{ account_key_passphrase | default(omit) | default(omit, true) }}"
    modify_account: false
    csr_content: "{{ retour_certreq.stdout }}"
    dest: "{{ data_dir }}/{{ certificats_nom }}.cer"
    challenge: http-01
    force: true
    terms_agreed: true
    account_email: "{{ contact_email }}"
  register: challenge_data

### ACME Etape 2 ################################################################################
- name: Récupération du certificate DIGICERT suite au challenge # noqa no-handler
  community.crypto.acme_certificate:
    select_crypto_backend: cryptography
    acme_version: 2
    acme_directory: "{{ acme_dir_url }}"
    account_key: "{{ data_dir }}/{{ account_key_name }}.pem"
    account_key_passphrase: "{{ account_key_passphrase | default(omit) | default(omit, true) }}"
    modify_account: false
    csr_content: "{{ retour_certreq.stdout }}"
    dest: "{{ data_dir }}/{{ certificats_nom }}.cer"
    challenge: http-01
    force: true
    terms_agreed: true
    account_email: "{{ contact_email }}"
    data: "{{ challenge_data }}"
  register: get_certificate_result
  when: challenge_data is changed
```

##### EXPECTED RESULTS

A CER file with :
- Common name :
test-123.domain.fr
- SANs :
test-123.domain.fr
test-123-1.domain.fr
test-123-2.domain.fr

##### ACTUAL RESULTS

Randomly CER file with sometimes :
- Common name :
test-123-1.domain.fr
- SANs :
test-123-1.domain.fr
test-123-2.domain.fr
test-123.domain.fr

Or
- Common name :
test-123-2.domain.fr
- SANs :
test-123-2.domain.fr
test-123.domain.fr
test-123-1.domain.fr

Or
- Common name :
test-123.domain.fr
- SANs :
test-123.domain.fr
test-123-1.domain.fr
test-123-2.domain.fr

```paste below

```

**Comment by felixfontein** [comment]
Using a list instead of a set would also be wrong:
1. The contract of that function is to return a set, not a list.
2. A list does not remove duplicates (like in case the same DNS name is present both as CNAME and SAN).

I'll take a closer look at this later.

**Comment by GuillaumeBernard3535** [comment]
But the return must be orderly, because the value of "cname" must remain in the first place. I'm sure you will find a solution :-)

**Comment by felixfontein** [comment]
I've created a fix in #725.
